File: app/src/main/java/com/ivanovsky/passnotes/data/entity/PropertyType.java
package com.ivanovsky.passnotes.data.entity;

import androidx.annotation.Nullable;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public enum PropertyType {
    TITLE("Title"),
    PASSWORD("Password"),
    USER_NAME("UserName"),
    URL("URL"),
    NOTES("Notes");

    public static Set<PropertyType> DEFAULT_TYPES = createDefaultPropertiesSet();
    private final String propertyName;

    private static Set<PropertyType> createDefaultPropertiesSet() {
        return new HashSet<>(Arrays.asList(TITLE, PASSWORD, USER_NAME, URL, NOTES));
    }

    @Nullable
    public static PropertyType getByName(@Nullable String name) {
        if (name == null) {
            return null;
        }

        for (PropertyType type : DEFAULT_TYPES) {
            if (type.propertyName.equalsIgnoreCase(name)) {
                return type;
            }
        }

        return null;
    }

    PropertyType(String propertyName) {
        this.propertyName = propertyName;
    }

    public String getPropertyName() {
        return propertyName;
    }
}

File: app/src/main/java/com/ivanovsky/passnotes/data/entity/PropertyType.kt
package com.ivanovsky.passnotes.data.entity

enum class PropertyType(val propertyName: String) {
    TITLE("Title"),
    PASSWORD("Password"),
    USER_NAME("UserName"),
    URL("URL"),
    NOTES("Notes"),
    OTP("otp");

    companion object {
        val DEFAULT_TYPES = setOf(TITLE, PASSWORD, USER_NAME, URL, NOTES)

        private val TYPES_MAP = mapOf(
            TITLE.propertyName.lowercase() to TITLE,
            PASSWORD.propertyName.lowercase() to PASSWORD,
            USER_NAME.propertyName.lowercase() to USER_NAME,
            URL.propertyName.lowercase() to URL,
            NOTES.propertyName.lowercase() to NOTES,
            OTP.propertyName.lowercase() to OTP
        )

        fun getByName(name: String?): PropertyType? {
            val loweredName = name?.lowercase() ?: return null

            return TYPES_MAP[loweredName]
        }
    }
}
